pandas/_libs/tslib.pyx

```diff
@@ -1268,13 +1266,33 @@ cdef class _Timestamp(datetime):
         # same timezone if specified)
         return datetime.__sub__(self, other)
 
-    cpdef _get_field(self, field):
+    cdef int64_t _maybe_convert_value_to_local(self):
+        """Convert UTC i8 value to local i8 value if tz exists"""
+        cdef:
+            int64_t val
         val = self.value
         if self.tz is not None and not _is_utc(self.tz):
             val = tz_convert_single(self.value, 'UTC', self.tz)
-        out = get_date_field(np.array([val], dtype=np.int64), field)
+        return val
+
+    cpdef _get_field(self, field):
+        cdef:
+            int64_t val
+            ndarray[int64_t] date_array = np.empty(1, dtype=np.int64)
+        val = self._maybe_convert_value_to_local()
+        date_array[0] = val
```

Comment:
this is overkill to type the input ndarray itself

it was ``out`` which was looking to type, though not a big deal, so just remove the date_array temp. ping on green.